[Justin Widodo] iP

GenFusion122 opened the pull request:
# Duck
> "Quack!" - Ducks, probably

## Have you ever wanted a simple planner/organizer? 
Well look no further! Duck is here to satisfy all your quacky needs.

Its,
- texted-based
- easy to use
- and angry all the time!

Installation steps:
1. Download the .jar from the Repo
2. Open it!
3. ~get abused by a duck!~

**its that simple!**

Features to be added:
- [x] Exception Handling
- [x] Save/Load Feature
- [x] GUI
- [ ] ~more angry duck noises~

Now here are some words from our sponsors: 
``` Java
System.out.println("It just works!");
System.out.println("Ducks are amazing!");
System.out.println("Quack!");
```

Commit:
Add Gradle support

GenFusion122 committed:
testing commit

GenFusion122 committed:
Renamed Duke to Duck
Completed Level-1

GenFusion122 committed:
Forgot to stage Duck.java

GenFusion122 committed:
Completed Level-2

GenFusion122 committed:
Completed Level-3

GenFusion122 committed:
Completed level-4

GenFusion122 committed:
Completed Level-5

GenFusion122 committed:
Completed Level-6

GenFusion122 committed:
added some comments as documentation

GenFusion122 committed:
Added Automated test cases

GenFusion122 committed:
Completed Level-7
Implemented a file save/load feature
Created an UnallowedCharacterException, to catch unallowed characters in user input

GenFusion122 committed:
Added some minor trimming to program

GenFusion122 committed:
Completed Level-8
Added date time to each object, rather than String

GenFusion122 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duck.java
#	src/main/java/Event.java

GenFusion122 committed:
Merged level 8 and 7 into master

GenFusion122 committed:
Merge branch 'master' of https://github.com/GenFusion122/CS2103-iP

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duck.java
#	src/main/java/Event.java

GenFusion122 committed:
Grouped everything into packages

GenFusion122 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Added gradle support

GenFusion122 committed:
completed A-JUnit

GenFusion122 committed:
Minor Changes to Gradle

GenFusion122 committed:
Added Javadocs to all functions

GenFusion122 committed:
Finished level-9

GenFusion122 committed:
Completed Level-10

phualiting commented on src/main/java/Commands/Commands.java:
```diff
@@ -0,0 +1,29 @@
+ */
+
+public abstract class Commands {
+
```
Should this extra line be removed?

phualiting commented on src/main/java/Commands/DeleteCommand.java:
```diff
@@ -0,0 +1,38 @@
+    private int index;
+
+    /**
+     *
```
Should this line also be deleted?

phualiting commented on src/main/java/Commands/DeleteCommand.java:
```diff
@@ -0,0 +1,38 @@
+     *
+     * @param index index of item to be deleted
+     */
+    public DeleteCommand(int index){
```
I think there is supposed to be a space before { for constructors?

phualiting commented on src/main/java/Commands/ExitCommand.java:
```diff
@@ -0,0 +1,35 @@
+ * Encapsulates the Exit or "bye" command
+ */
+public class ExitCommand extends Commands{
+    public ExitCommand(){
```
Not very sure about the best way of writing empty constructors, but it might be better to standardize all of them to ExitCommand() {}

phualiting commented on src/main/java/Commands/FindCommand.java:
```diff
@@ -0,0 +1,54 @@
+    public void execute(TaskList<Todo> list, Storage storage, UI ui) throws IndexOutOfBoundsException {
+        boolean foundAny = false;
+        String s = "";
+        for (int i = 0; i < list.size(); i++){
```
There should be a space before {?

GenFusion122 replied:
I blame late night coding, thanks!

phualiting commented on src/main/java/Commands/FindCommand.java:
```diff
@@ -0,0 +1,54 @@
+package Commands;
+
```
I think it would be better to standardize if there is a line separating package and imports

phualiting commented on src/main/java/Duck/Parser.java:
```diff
@@ -0,0 +1,87 @@
+                String[] arr = input.split(" ", 2);
+                String command = arr[0];
+                String arguments = arr[1];
+                switch (command.toUpperCase()) {
+                    case "TODO":
+                        Todo newTodo = new Todo(arguments, false);
+                        return new AddCommand(newTodo);
+                    case "DEADLINE":
+                        String[] deadlineArgs = arguments.split("/by");
+                        Deadline newDeadline = new Deadline(deadlineArgs[0].trim(), false, Duck.dateStorageConverter(deadlineArgs[1].trim()));
+                        return new AddCommand(newDeadline);
+                    case "EVENT":
+                        String[] eventArgs = arguments.split("/at");
+                        Event newEvent = new Event(eventArgs[0].trim(), false, Duck.dateStorageConverter(eventArgs[1].trim()));
+                        return new AddCommand(newEvent);
+                    case "MARK":
+                        return new MarkCommand(Integer.parseInt(arguments) - 1);
+                    case "UNMARK":
+                        return new UnmarkCommand(Integer.parseInt(arguments) - 1);
+                    case "DELETE":
+                        return new DeleteCommand(Integer.parseInt(arguments) - 1);
+                    case "FIND":
+                        return new FindCommand(arguments);
+                    default:
+                        ui.sendTextToUi("Quack! What does that even mean ?!?!?");
+                        break;
```
Switch and case should have same indentation?

GenFusion122 replied:
Not sure how i missed that out, thanks!

phualiting commented on src/main/java/Duck/Storage.java:
```diff
@@ -0,0 +1,77 @@
+package Duck;
+import java.io.*;
```
Should list all imported classes explicitly

GenFusion122 replied:
Intellij loves to collate the imports to a *, i've changed the settings! Thanks for pointing it out!

phualiting reviewed (commented):
Overall, I like how Javadocs were clearly written for most classes and methods. However, there were a few coding standard errors, and I think you can consider making use of checkstyle to easily spot those violations.

GenFusion122 committed:
Rectified minor coding standard violations in the following classes:
Commands.java
DeleteCommand.java
ExitCommand.java
FindCommand.java
Parser.java
Storage.java

GenFusion122 committed:
Include assertions in various part of the code

Currently, there are no assertions to verify certain object
creations in the code

Adding assertations will help with bug fixing

Assertions are useful because they are easily disabled and do not
take up much resources to run/execute

GenFusion122 committed:
Merge pull request #1 from GenFusion122/branch-A-Assertions

Include assertions in various part of the code

GenFusion122 committed:
Refactor code to align with quality coding practices

Following quality coding practices allows for increased readability
of the code and therefore, making improving its content

Several classes have been refactored to a higher coding standard.
Typically, none of the functionalities were changed.

GenFusion122 committed:
Merge pull request #2 from GenFusion122/branch-A-CodeQuality

Refactor code to align with quality coding practices

GenFusion122 committed:
Completed BCD-Extension - Specifically, C-Update

GenFusion122 committed:
Refactor code to fit coding standards, defined in email sent

GenFusion122 committed:
README Update

GenFusion122 committed:
Update README.md

GenFusion122 committed:
Upload Ui.png

GenFusion122 committed:
Merge branch 'master' of https://github.com/GenFusion122/CS2103-iP

GenFusion122 committed:
Update correct README.md

GenFusion122 committed:
Update README.md
